r/vignettes/install.Rmd

```diff
@@ -379,63 +312,86 @@ setting `ARROW_WITH_ZSTD=OFF` to build without `zstd`; or (3) uninstalling
+| `ARROW_WITH_LZ4`    | Compression algorithm  | `ON` |
+| `ARROW_WITH_SNAPPY`    | Compression algorithm  | `ON` |
+| `ARROW_WITH_ZLIB`    | Compression algorithm | `ON` |
+| `ARROW_WITH_ZSTD`    | Compression algorithm | `ON` |
+
+## R package configuration
+
+There are a number of other variables that affect the `configure` script and 
+the bundled build script.  All boolean variables are case-insensitive.
+
+| Name | Description | Default |
+| --- | --- | :-: |
+| `ARROW_USE_PKG_CONFIG` | Use `pkg-config` to search for `libarrow` install | `true` |
+| `LIBARROW_BUILD` | Allow building from source | `true` |
+| `LIBARROW_BINARY` | Try to install `libarrow` binary instead of building from source | `true` |
+| `LIBARROW_MINIMAL` | Build with minimal features enabled | (unset) |
+| `NOT_CRAN` | Set `LIBARROW_BINARY=true` and `LIBARROW_MINIMAL=false` | `false` |
+| `ARROW_R_DEV` | More verbose messaging and regenerates some code | `false` |
+| `LIBARROW_DEBUG_DIR` | Directory to save source build logs | (unset) |
+| `CMAKE` | Alternative CMake path | (unset) |
+
+See below for more in-depth explanations of these environment variables.
+
+* `ARROW_USE_PKG_CONFIG`: If set to `false`, the configure script won't look for 
+Arrow libraries on your system and instead will look to download/build them.
+  Use this if you have a version mismatch between installed system libraries and
+  the version of the R package you're installing.
+* `LIBARROW_BINARY` : If set to `true`, the script will try to download a binary
+  C++ library built for your operating system.  You may also set it to some other string,  a related "distro-version" that has binaries built that work for your OS.
+  If no binary is found, installation will fall back to building C++  dependencies from source. 
+* `LIBARROW_BUILD` : If set to `false`, the build script
   will not attempt to build the C++ from source. This means you will only get
   a working `arrow` R package if a prebuilt binary is found.
   Use this if you want to avoid compiling the C++ library, which may be slow
-  and resource-intensive, and ensure that you only use a prebuilt binary.
-* `LIBARROW_MINIMAL`: If set to `false`, the build script
-  will enable some optional features, including compression libraries, S3
-  support, and additional alternative memory allocators. This will increase the
-  source build time but results in a more fully functional library.
-* `NOT_CRAN`: If this variable is set to `true`, as the `devtools` package does,
+  and resource-intensive, and ensure that you only use a prebuilt binary. 
+* `LIBARROW_MINIMAL` : If set to `false`, the build script
+  will enable some optional features, including S3
+  support and additional alternative memory allocators. This will increase the
+  source build time but results in a more fully functional library. If set to 
+  `true` turns off Parquet, Datasets, compression libraries, and other optional 
+  features.
```

Comment:
```suggestion
  features (which is only really useful if compiling on a platform that does not support these features, e.g. Solaris)
```

This might be too much snark, but it _is_ the reason for it.

Reply:
Added a less snarky rephrasing of this!